Taxonomy subtree deletion and clean-up (#2271)
ibexa/documentation-developer: docs/content_management/taxonomy/taxonomy.md

```

For more information about available functionalities of tags, see [User Documentation]([[= user_doc =]]/taxonomy).

## Hide delete button on large subtree

> The "Delete" button can be hidden when a taxonomy entry has many children. By default, the entry's "Delete" button is hidden when there are 100 children or more.

Reviewer: ```suggestion
The **Delete** button can be hidden when a taxonomy entry has many children.
By default, the button is hidden when there are 100 children or more.
```

The `delete_subtree_size_limit` configuration is [SiteAccess-aware](siteaccess_aware_configuration.md), and can be set per SiteAccess, per SiteAccess group, or globally per default.
For example:

```yaml
ibexa:
    system:
        default: # or a SiteAccess, or a SiteAccess group
            taxonomy:
                admin_ui:
                    delete_subtree_size_limit: 100
```

## Remove orphaned Content items

> In some rare case, especially in Ibexa DXP 4.2 and older versions, when deleting parent of huge subtrees, some Taxonomy entries are not properly deleted, leaving Content items pointing a non-existing parent.

Reviewer: ```suggestion
In some rare case, especially in I[[= product_name =]] v4.2 and older, when deleting parent of huge subtrees, some Taxonomy entries are not properly deleted, leaving Content items that point to a non-existing parent.
```

> The command `ibexa:taxonomy:remove-orphaned-content` delete those orphaned Content item. This command works on one taxonomy passed as an argument. It has two options, one should always be used:

Reviewer: ```suggestion
The command `ibexa:taxonomy:remove-orphaned-content` deletes those orphaned Content item. 
It works on a taxonomy passed as an argument, and has two options that act as a protective measure against deleting data by mistake:
```

> - `--dry-run` to list Content items that will be deleted without performing the deletion.

Reviewer: ```suggestion
- `--dry-run` to list Content items that will be deleted, without performing the deletion.
```

- `--force` to effectively delete the orphaned Content items.

> The following example,
> first, list the orphaned Content items for taxonomy `tags`,
> second, delete them:

Reviewer: ```suggestion
The following example first lists the orphaned Content items for taxonomy `tags`,
and then deletes them:
```

```

For more information about available functionalities of tags, see [User Documentation]([[= user_doc =]]/taxonomy).

## Hide delete button on large subtree

The **Delete** button can be hidden when a taxonomy entry has many children.
By default, the button is hidden when there are 100 children or more.

The `delete_subtree_size_limit` configuration is [SiteAccess-aware](siteaccess_aware_configuration.md), and can be set per SiteAccess, per SiteAccess group, or globally per default.
For example:

```yaml
ibexa:
    system:
        default: # or a SiteAccess, or a SiteAccess group
            taxonomy:
                admin_ui:
                    delete_subtree_size_limit: 20
```

## Remove orphaned Content items

In some rare case, especially in I[[= product_name =]] v4.2 and older, when deleting parent of huge subtrees, some Taxonomy entries are not properly deleted, leaving Content items that point to a non-existing parent.

> The command `ibexa:taxonomy:remove-orphaned-content` deletes those orphaned Content item. 

Author: ```suggestion
The command `ibexa:taxonomy:remove-orphaned-content` deletes those orphaned Content item.
```

It works on a taxonomy passed as an argument, and has two options that act as a protective measure against deleting data by mistake:

> - `--dry-run` to list Content items that will be deleted, without performing the deletion.

Author: ```suggestion
- `--dry-run` to list Content items that would be deleted, without performing the deletion.
```

Author: ```suggestion
- `--dry-run` to list deletable Content items, without performing the deletion.
```

- `--force` to effectively delete the orphaned Content items.

> The following example first lists the orphaned Content items for taxonomy `tags`,
> and then deletes them:

Reviewer: ```suggestion
The following example first lists the orphaned Content items for taxonomy `tags`, and then deletes them:
```
